api/src/main/java/org/apache/cloudstack/api/command/user/network/MoveNetworkAclItemCmd.java

```diff
@@ -43,6 +43,9 @@
     @Parameter(name = ApiConstants.NEXT_ACL_RULE_ID, type = CommandType.STRING, description = "The ID of the rule that is right after the new position where the rule being moved is going to be placed. This value can be 'NULL' if the rule is being moved to the last position of the network ACL list.")
     private String nextAclRuleUuid;
 
+    @Parameter(name = ApiConstants.MOVE_ACL_CONSISTENCY_HASH, type = CommandType.STRING, description = "Md5 hash used to check the consistency of the ACL rule list before applying the ACL rule move. This check is useful to manage concurrency problems that may happen when multiple users are editing the same ACL rule listing. The parameter is not required. Therefore, if the user does not send it, he/she is assuming the risk of moving ACL rules without checking the consistency of the access control list before executing the move. We use MD5 hash function on a String that is composed of all UUIDs of the ACL rules in concatenated in their respective order (order defined via 'number' field).")
```

Comment:
very solid method but very hard to use when calling the API, isn't it?
Would this work or what could I do with for instance cloudmonkey?

Reply:
Yes, you are right. I tried to think of a better solution, but I could not come up with anything better: (

I do not see much difficulty to use it programmatically. However, I do see the problem with cloudmonkey or other similar tools if they do not provide some out of the box enabler methods.  On the bright side, ACS UI works like a charm without any burden to operators or users.

BTW: I am always open to suggestions!

Reply:
Ah, there is something else that I forgot to mention. The hash is not mandatory. Therefore, it will work via CloudMonkey, but it will not be a safe operation (as it was not in the past).